src/DssDirectDepositHub.sol

```diff
@@ -298,39 +305,41 @@ contract DssDirectDepositHub {
             );
         } else {
             // Normal path
+
+            // Determine if it needs to unwind due debt ceilings
+            (uint256 Art,,, uint256 line,) = vat.ilks(ilk_);
+            uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+            uint256 Line = vat.Line();
+            uint256 debt = vat.debt();
+            uint256 toUnwind;
+            if (Art > lineWad) {
+                toUnwind = Art - lineWad;
+            }
+            if (debt > Line) {
+                toUnwind = _max(toUnwind, _divup(debt - Line, RAY));
+            }
```

Comment:
how would you feel about moving this to an internal function to encapsulate this specific logic?

Reply:
i'm fine with leaving as-is